### PR TITLE
Hands On 2 GROUP11

### PR DESCRIPTION
**Pull request by vcuende**

### CODE REVIEW

File: Assignment4/vcuende-180188/task06.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1L28eRU7IB0QsXz5d8h4hKsf-Bqc4exdt
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/vcuende-180188/task06.py:12]: This won't run in Python. You need to comment this line. The same applies for the other tasks

File: Assignment4/vcuende-180188/task07.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+from rdflib.plugins.sparql import prepareQuery
+
+NS = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    {?Subject rdf:type ns:Person}
+    UNION
+    {?Subject rdf:type ns:Researcher}
```
Comment (dgarijo) [inline Assignment4/vcuende-180188/task07.py:53]: You need to do this query in a generic manner, not listing all the classes that you know are subClassOf Person

File: Assignment4/vcuende-180188/task07.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+from rdflib.plugins.sparql import prepareQuery
+
+NS = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?Subject ?Prop ?X WHERE { 
+    ?Subject rdf:type ns:Person.
+    ?Subject ?Prop ?X.
+
```
Comment (dgarijo) [inline Assignment4/vcuende-180188/task07.py:74]: This is missing all the elements from all the instances of subclasses of person

File: Assignment4/vcuende-180188/task07.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    ?Subject rdf:type ns:Person.
+    ?Subject ?Prop ?X.
+    
+  }
+  ''',
+  initNs = { "ns": NS}
+)
+
+for r in g.query(q1):
+  print(r.Subject, r.Prop)
```
Comment (dgarijo) [inline Assignment4/vcuende-180188/task07.py:81]: Missing how to do 7.1, 7.2 and 7.3 in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
There are a few errors and exercises missing. Please complete them

File: Assignment4/vcuende-180188/task07.py (hunk 4 above)
Comment (vcuende) [reply Assignment4/vcuende-180188/task07.py]: Fixed

File: Assignment4/vcuende-180188/task07.py (hunk 3 above)
Comment (vcuende) [reply Assignment4/vcuende-180188/task07.py]: Fixed

File: Assignment4/vcuende-180188/task07.py (hunk 2 above)
Comment (vcuende) [reply Assignment4/vcuende-180188/task07.py]: Fixed

File: Assignment4/vcuende-180188/task06.py (hunk 1 above)
Comment (vcuende) [reply Assignment4/vcuende-180188/task06.py]: Fixed

**Pull request closed without merging**